frontend/app/utils/route-utils.ts

```diff
@@ -131,26 +131,29 @@ export function usePageTitleI18nOptions() {
     .reduce(coalesce);
 }
 
-export function findRouteById(id: string, routes: I18nRoute[] = []): I18nPageRoute | undefined {
-  for (const route of routes) {
-    if (isI18nPageRoute(route) && route.id === id) {
-      return route;
-    }
-
-    if (isI18nLayoutRoute(route)) {
-      const matchingRoute = findRouteById(id, route.children);
-      if (matchingRoute) return matchingRoute;
+export function findRouteById(id: string): I18nPageRoute | undefined {
+  function search(id: string, routes: I18nRoute[]): I18nPageRoute | undefined {
```

Comment:
What's the purpose and benefit of `search` function inside `findRouteById`.

Reply:
``` typescript
export function findRouteById(id: string): I18nPageRoute | undefined {
  function search(id: string, routes: I18nRoute[]): I18nPageRoute | undefined {
    for (const route of routes) {
      if (isI18nPageRoute(route) && route.id === id) {
        return route;
      }

      if (isI18nLayoutRoute(route)) {
        const matchingRoute = search(id, route.children);
        if (matchingRoute) return matchingRoute;
      }
    }
  }

  return search(id, routes);
}
```

I didn't want to have to call `findRouteById()` with a `routes` parameter every time.. but I needed a recursive function because a route can have children...

But now that I look at it closer, I think I can just specify `routes` as the default value. I'll modify it and push a commit.

Reply:
https://github.com/DTS-STN/canadian-dental-care-plan/pull/2566/commits/45f631a89d299d74103ed8a383c62256bf6778f1